apps/src/templates/sectionProgress/progressTables/ProgressTableLevelBubble.jsx

```diff
@@ -68,6 +68,130 @@ const styles = {
   }
 };
 
+function levelStyle(props) {
+  const {levelStatus, levelKind, disabled} = props;
+  return {
+    ...(!disabled && progressStyles.hoverStyle),
+    ...progressStyles.levelProgressStyle(levelStatus, levelKind, disabled)
+  };
+}
+
+function mainStyle(props) {
+  return {
+    ...styles.main,
+    ...levelStyle(props)
+  };
+}
+
+function largeStyle(props) {
+  const {disabled, bonus} = props;
+  return {
+    ...mainStyle(props),
+    ...styles.large,
+    ...(disabled && bonus && styles.bonusDisabled)
+  };
+}
+
+function largeContentStyle(props) {
+  const {bonus, paired} = props;
+  return {
+    ...progressStyles.flex,
+    ...styles.contents,
+    fontSize: paired || bonus ? 14 : 16
+  };
+}
+
+function UnpluggedBubble(props) {
+  return (
+    <div
+      style={{
+        ...styles.main,
+        ...styles.unplugged,
+        ...levelStyle(props)
+      }}
+    >
+      <Content {...props} />
+    </div>
+  );
+}
+
+function SmallCircle(props) {
+  return (
+    <div>
+      <div style={{...mainStyle(props), ...styles.smallCircle}}>
+        <Content {...props} />
+      </div>
+    </div>
+  );
+}
+
+function LargeBubble(props) {
+  const BubbleType = props.concept ? LargeDiamond : LargeCircle;
+  return (
+    <div style={styles.container}>
+      <BubbleType {...props} />
+    </div>
+  );
+}
+LargeBubble.propTypes = {
+  concept: PropTypes.bool
+};
+
+function LargeCircle(props) {
+  return (
+    <div style={{...largeStyle(props), ...styles.largeCircle}}>
+      <div style={largeContentStyle(props)}>
+        <Content {...props} />
+      </div>
+    </div>
+  );
+}
+
+function LargeDiamond(props) {
+  return (
+    <div style={{...largeStyle(props), ...styles.largeDiamond}}>
+      <div style={{...largeContentStyle(props), ...styles.diamondContents}}>
+        <Content {...props} />
+      </div>
+    </div>
+  );
+}
+
+function Content(props) {
+  const {levelStatus, unplugged, bonus, paired, title} = props;
+  const locked = levelStatus === LevelStatus.locked;
+  return unplugged ? (
+    <span>{i18n.unpluggedActivity()}</span>
+  ) : locked ? (
+    <FontAwesome icon="lock" />
+  ) : paired ? (
+    <FontAwesome icon="users" />
+  ) : bonus ? (
+    <FontAwesome icon="flag-checkered" />
+  ) : title ? (
+    <span>{title}</span>
+  ) : null;
+}
+Content.propTypes = {
+  levelStatus: PropTypes.string,
+  unplugged: PropTypes.bool,
+  bonus: PropTypes.bool,
+  paired: PropTypes.bool,
+  title: PropTypes.string
+};
+
+function LinkWrapper(props) {
```

Comment:
I like the different bubble types factored out, I think that makes sense to me. This wrapper doesn't seem super valuable outside of testing, especially since it's only used in one place. IMO the code would feel more readable if we just wrapped the return in a link.

Reply:
see my comment below in the tests and lmk what you think